app/main.py

```diff
@@ -1,2 +1,63 @@
+from typing import Any
+
+
+class Node:
+    def __init__(self, key: Any, value: Any) -> None:
+        self.key = key
+        self.value = value
+        self.hash = hash(key)
+
+
 class Dictionary:
-    pass
+    def __init__(self) -> None:
+        self.capacity = 8
+        self.size = 0
+        self.load_factor = 2 / 3
+        self.hash_table = [None] * self.capacity
+
+    def __len__(self) -> int:
+        return self.size
+
+    def resize(self) -> None:
+        self.capacity *= 2
+        self.size = 0
+        old_table = self.hash_table
+        self.hash_table = [None] * self.capacity
+
+        for node in old_table:
+            if node is not None:
+                self.__setitem__(node.key, node.value)
+
+    def __setitem__(self, key: Any, value: Any) -> None:
+        index = hash(key) % self.capacity
```

Comment:
Currently, resizing happens after inserting a new element when the load factor exceeds 2/3. But this means for every new insertion past the limit, an unnecessary resize might occur, especially for high-load situations.

Perform the resize before the insertion to avoid inserting and then resizing immediately.

```
 if self.size / object > object:  # Check before insertion
        self.resize()
```